applications/GeoMechanicsApplication/custom_utilities/solving_strategy_factory.hpp

```diff
@@ -44,6 +44,11 @@ class SolvingStrategyFactory
         if (rSolverSettings[strategy_type].GetString() ==
             GeoMechanicsNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::Name())
         {
+            const int max_iterations = rSolverSettings["max_iterations"].GetInt();
+            const bool calculate_reactions = rSolverSettings["calculate_reactions"].GetBool();
+            const bool reform_dof_set_at_each_step = rSolverSettings["reform_dofs_at_each_step"].GetBool();
+            const bool move_mesh_flag = rSolverSettings["move_mesh_flag"].GetBool();
```

Comment:
I would use `const auto` here for all cases, since the types are clear from the expressions that initialize the variables:
```suggestion
            const auto max_iterations = rSolverSettings["max_iterations"].GetInt();
            const auto calculate_reactions = rSolverSettings["calculate_reactions"].GetBool();
            const auto reform_dof_set_at_each_step = rSolverSettings["reform_dofs_at_each_step"].GetBool();
            const auto move_mesh_flag = rSolverSettings["move_mesh_flag"].GetBool();
```